File: M4 Intermediate DSA/src/L13_Maths_Modular_Arthimetic_Introduction/Assignment2.java
package L13_Maths_Modular_Arthimetic_Introduction;

public class Assignment2 {

  public int solve(int[] A, int B) {
    long ans = 0;
    int N = A.length;
    long pow = 1;

    for (int index = N -1; index >= 0; index--) {
      int digit = A[index];
      long term = (digit * pow) % B;
      ans = ((ans % B) + (term % B)) % B;
      pow = (pow * 10) % B;
    }

    return (int)ans;
  }

}

File: M4_Intermediate_DSA/L2_Modular_Arthimetic_Introduction/assignment2.go
package main

func SolveA2(A []int, B int) int {
	ans := 0
	pow := 1

	for i := len(A) - 1; i >= 0; i-- {
		digit := A[i]
		term := ((digit % B) * (pow % B)) % B
		ans = ((ans % B) + (term % B)) % B
		pow = ((pow % B) * (10 % B)) % B
	}

	return ans
}
